### DISCUSSION TITLE
Profiling

### DISCUSSION THREAD

**Discussion by hinell** [shellRaining/hlchunk.nvim#81] [Ideas]
@shellRaining Hi.

Just wanted to let you know that there is a [built-in LuaJit Profiler.](https://luajit.org/ext_profiler.html) you can use for profiling huge files .e.g how I did it:

1. Run in shell
    ```bash
    export PROFILE_FILENAME=hlchunk.profile.txt
    nvim PATH_TO_BIG_FILE && nvim -O $PROFILE_FILENAME
    ```
2. Then, run in nvim's cmdline:
    ```lua
    lua require"jit.p".start("10,i1,s,m0", vim.fn.getcwd() .. "/" .. vim.env.PROFILE_FILENAME)
    ``` 

3. Run when you finish testing:
    ```lua
    lua require"jit.p".stop()
    ```
4. Exit window. It will open `PROFILE_FILENAME` automatically.

My best.

**Comment by shellRaining** [comment]
Thank you very much for the information! I will take a look. but I am facing graduation issues recently🥲, I may spend less time in this repo. and dev branch is such a mess that I want to speed more time on it.

**Reply by hinell** [reply]
@shellRaining No worries. Just saying! Cheers.

**Comment by shellRaining** [comment]
now using `profile.nvim` to profile huge files, and I think that the performance is fine now.
